app/src/main/java/com/spendster/presentation/signUp/ServerSignUpModel.java

```diff
@@ -1,14 +1,14 @@
 package com.spendster.presentation.signUp;
 
 import com.spendster.data.entity.User;
+import com.spendster.presentation.APIClient;
 
 import io.reactivex.Single;
 
 public class ServerSignUpModel implements SignUpModel {
-
     @Override
-    public Single<User> getUser(String email, String password) {
-        API api = APIClient.getClient().create(API.class);
-        return api.signUp(new User(email, password));
+    public Single<User> getUser(String email, String username, String password) {
+        APISignUp apiSignUp = APIClient.getClient().create(APISignUp.class);
```

Comment:
@SimonMaha let's extract apiSignup into constructor and pass it here from outside.